lisp/modules/rename_cues/rename_ui.py

```diff
@@ -0,0 +1,299 @@
+            if cue['selected']:
+                cue['cue_preview'] = cue['cue_preview'].title()
+        self.update_preview_list()
+
+    def onLowerButtonClicked(self):
+        for cue in self._cues_list:
+            if cue['selected']:
+                cue['cue_preview'] = cue['cue_preview'].lower()
+        self.update_preview_list()
+
+    def onUpperButtonClicked(self):
+        for cue in self._cues_list:
+            if cue['selected']:
+                cue['cue_preview'] = cue['cue_preview'].upper()
+        self.update_preview_list()
+
+    def onRemoveNumButtonClicked(self):
+        regex = re.compile('^[^a-zA-Z]+(.+)')
+
+        def remove_numb(input):
+            match = regex.search(input)
+            if match is not None:
+                return match.group(1)
+            else:
+                return input
+
+        for cue in self._cues_list:
+            if cue['selected']:
+                cue['cue_preview'] = remove_numb(cue['cue_preview'])
+
+        self.update_preview_list()
+
+    def onAddNumberingButtonClicked(self):
+        # Extract selected rows
+        cues_to_modify = [cue for cue in self._cues_list if cue['selected']]
+        # Count selected rows
+        cues_nbr = len(cues_to_modify)
+        # Calculate number of digits in order to add appropriate number of 0's
+        digit_nbr = len(str(cues_nbr))
+
+        for i, cue in enumerate(cues_to_modify):
+            if cue['selected']:
+                cue['cue_preview'] = f"{i+1:0{digit_nbr}.0f} - {cue['cue_preview']}"
+
+        self.update_preview_list()
+
+    def onResetButtonClicked(self):
+        for cue in self._cues_list:
+            cue['cue_preview'] = cue['cue_name']
+            cue['selected'] = True
+
+        self.update_preview_list()
+
+    def onRegexLineChanged(self):
+        pattern = self.regexLine.text()
+        try:
+            regex = re.compile(pattern)
+        except re.error:
+            logging.info("Regex error : not a valid pattern")
+        else:
+            for cue in self._cues_list:
+                result = regex.search(cue['cue_name'])
+                if result:
+                    cue['regex_groups'] = result.groups()
+
+            self.onOutRegexChanged()
+
+    def onOutRegexChanged(self):
+        out_pattern = self.outRegexLine.text()
+
+        if out_pattern == '':
+            for cue in self._cues_list:
+                if cue['selected']:
+                    cue['cue_preview'] = cue['cue_name']
+        else:
+            for cue in self._cues_list:
+                out_string = out_pattern
+                for n in range(len(cue['regex_groups'])):
+                    pattern = f"\${n}"
+                    try:
+                        out_string = re.sub(pattern,
+                                            cue['regex_groups'][n], out_string)
+                    except IndexError:
+                        logging.info("Regex error : Catch with () before display with $n")
```

Comment:
Ok. I thought an error should never pass silently.... Are we sure that no other IndexError is possible here ? Or should we put it in debug level ?